src/ziggurat/streams.clj

```diff
@@ -160,6 +180,50 @@
       (finally
         (.finish span)))))
 
+(defn- join-streams
+  [oldest-processed-message-in-s handler-fn channels topic-entity stream-1 stream-2]
+  (let [topic-entity-name (name topic-entity)
+        topic-1           (:topic-name stream-1)
+        topic-2           (:topic-name stream-2)
+        cfg-1             (:cfg stream-1)
+        cfg-2             (:cfg stream-2)
+        strm-1            (->> (:stream stream-1)
+                               (stream-joins-delay-metric topic-1 topic-entity-name oldest-processed-message-in-s)
+                               (map-values #(stream-joins-log-and-report-metrics topic-1 topic-entity-name %)))
+        strm-2            (->> (:stream stream-2)
+                               (stream-joins-delay-metric topic-2 topic-entity-name oldest-processed-message-in-s)
+                               (map-values #(stream-joins-log-and-report-metrics topic-2 topic-entity-name %)))
+        join-window       (JoinWindows/of (Duration/ofMillis (:join-window cfg-1)))
+        join-type         (:join-type cfg-1)
+        value-joiner      (reify ValueJoiner
+                            (apply [_ left right]
+                              {:left left :right right}))
+        out-strm          (->> (if cfg-1
+                                 (case join-type
+                                   :left  (.leftJoin strm-1 strm-2 value-joiner join-window)
+                                   :outer (.outerJoin strm-1 strm-2 value-joiner join-window)
+                                   (.join strm-1 strm-2 value-joiner join-window))
+                                 strm-1)
+                               (header-transform-values)
+                               (map-values #(traced-handler-fn handler-fn channels % topic-entity)))]
+    {:stream out-strm
+     :cfg    cfg-2}))
+
+(defn- create-stream [builder topic cfg]
+  (let [stream (case (:input-type cfg)
+                 ;; :stream (.stream builder topic)
```

Comment:
remove commented out code

Reply:
can we leave it here as a placeholder when we support ktables?

Reply:
removed...